bench/bench_test.go

```diff
@@ -15,41 +15,50 @@ import (
 )
 
 func BenchmarkEngines(b *testing.B) {
-	buf, _ := os.ReadFile("case/case.wasm")
 	b.Run("naivevm", func(b *testing.B) {
 		store := newStore(naivevm.NewEngine())
-		mod, err := wasm.DecodeModule((buf))
-		if err != nil {
-			panic(err)
-		}
-		err = store.Instantiate(mod, "test")
-		if err != nil {
-			panic(err)
-		}
-		runBase64Benches(b, store)
-		runFibBenches(b, store)
-		runStringsManipulationBenches(b, store)
-		runReverseArrayBenches(b, store)
-		runRandomMatMul(b, store)
+		setUpStore(store)
+		runAllBenches(b, store)
 	})
 	b.Run("wazeroir", func(b *testing.B) {
 		store := newStore(wazeroir.NewEngine())
-		mod, err := wasm.DecodeModule((buf))
-		if err != nil {
-			panic(err)
-		}
-		err = store.Instantiate(mod, "test")
-		if err != nil {
-			panic(err)
-		}
-		runBase64Benches(b, store)
-		runFibBenches(b, store)
-		runStringsManipulationBenches(b, store)
-		runReverseArrayBenches(b, store)
-		runRandomMatMul(b, store)
+		setUpStore(store)
+		runAllBenches(b, store)
 	})
 }
 
+func setUpStore(store *wasm.Store) {
+	buf, err := os.ReadFile("case/case.wasm")
+	if err != nil {
+		panic(err)
+	}
+	mod, err := wasm.DecodeModule((buf))
+	if err != nil {
+		panic(err)
+	}
+	err = store.Instantiate(mod, "test")
+	if err != nil {
+		panic(err)
+	}
+
+	// We assume that TinyGo binary expose "_start" symbol
```

Comment:
is there an api to check existence of a function? if so, we could still do this, and work if we switch the wasm implementation.

Reply:
we don't have such API but the CallFunction returns error in these case. I would create specific error for these cases `ErrFunctionNotFound`, but leave it for other PRs!

Reply:
https://github.com/tetratelabs/wazero/blob/main/wasm/store.go#L162-L170

Reply:
I will raise a PR for `HasFunction` or similar!